### ISSUE TITLE
Very noisy logging during create/install

### ISSUE THREAD

**Issue by jakirkham** [conda/conda#3236]
Getting all sorts of logging noisy from `conda` when installing packages or creating environments. Info and an example of the noise provided below.

```
$ conda info
Current conda install:

             platform : osx-64
        conda version : 4.1.11
    conda-env version : 2.5.2
  conda-build version : 1.21.9
       python version : 2.7.11.final.0
     requests version : 2.10.0
     root environment : /zopt/conda2  (writable)
  default environment : /zopt/conda2/envs/nanshenv
     envs directories : /zopt/conda2/envs
        package cache : /zopt/conda2/pkgs
         channel URLs : https://conda.anaconda.org/t/<TOKEN>/conda-forge/osx-64/
                        https://conda.anaconda.org/t/<TOKEN>/conda-forge/noarch/
                        https://conda.anaconda.org/t/<TOKEN>/nanshe/osx-64/
                        https://conda.anaconda.org/t/<TOKEN>/nanshe/noarch/
                        https://repo.continuum.io/pkgs/free/osx-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/osx-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
                        https://conda.anaconda.org/t/<TOKEN>/jakirkham/osx-64/
                        https://conda.anaconda.org/t/<TOKEN>/jakirkham/noarch/
                        https://conda.anaconda.org/t/<TOKEN>/javascript/osx-64/
                        https://conda.anaconda.org/t/<TOKEN>/javascript/noarch/
          config file : /Users/kirkhamj/.condarc
         offline mode : False
    is foreign system : False

$ conda list -n root
# packages in environment at /zopt/conda2:
#
Using Anaconda API: https://api.anaconda.org
conda                     4.1.11                   py27_0    defaults
conda-build               1.21.9                   py27_0    defaults
anaconda-client           1.5.1                    py27_0    conda-forge
anaconda-verify           1.2.0                    py27_0    conda-forge
ca-certificates           2016.8.2                      3    conda-forge
certifi                   2016.8.2                 py27_0    conda-forge
clyent                    1.2.1                    py27_0    conda-forge
conda-build-all           0.13.1                   py27_0    conda-forge
conda-env                 2.5.2                    py27_0    conda-forge
conda-execute             0.6.0                    py27_0    conda-forge
curl                      7.49.1                        0    conda-forge
expat                     2.1.0                         1    conda-forge
funcsigs                  1.0.2                    py27_0    conda-forge
gitdb                     0.6.4                    py27_0    conda-forge
gitpython                 1.0.1                    py27_0    conda-forge
jinja2                    2.8                      py27_1    conda-forge
libconda                  4.0.0                    py27_0    conda-forge
markupsafe                0.23                     py27_0    conda-forge
mock                      2.0.0                    py27_0    conda-forge
ncurses                   5.9                           8    conda-forge
openssl                   1.0.2h                        2    conda-forge
pip                       8.1.2                    py27_0    conda-forge
psutil                    4.3.0                    py27_0    conda-forge
pycosat                   0.6.1                    py27_0    conda-forge
pycrypto                  2.6.1                    py27_0    conda-forge
pygithub                  1.26.0                   py27_0    conda-forge
python-dateutil           2.5.3                    py27_0    conda-forge
pytz                      2016.6.1                 py27_0    conda-forge
pyyaml                    3.11                     py27_0    conda-forge
readline                  6.2                           0    conda-forge
requests                  2.10.0                   py27_0    conda-forge
ruamel.ordereddict        0.4.6                    py27_0    conda-forge
ruamel.yaml               0.11.11                  py27_1    conda-forge
ruamel_yaml               0.11.7                   py27_0    conda-forge
setuptools                25.1.3                   py27_0    conda-forge
six                       1.10.0                   py27_0    conda-forge
smmap                     0.9.0                    py27_0    conda-forge
sqlite                    3.13.0                        1    conda-forge
tk                        8.5.19                        0    conda-forge
wheel                     0.29.0                   py27_0    conda-forge
yaml                      0.1.6                         0    conda-forge
zlib                      1.2.8                         3    conda-forge
constructor               1.3.1                    py27_0    defaults
pbr                       1.10.0                   py27_0    defaults
python                    2.7.11                        0    defaults

$ conda create -y -n numpyenv numpy
Using Anaconda API: https://api.anaconda.org
Fetching package metadata ...INFO:stdoutlog:Fetching package metadata ...
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): conda.anaconda.org
.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/t/ja-1ba1ac63-9f8d-450e-a469-83216f398a2f/conda-forge/osx-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/t/ja-1ba1ac63-9f8d-450e-a469-83216f398a2f/conda-forge/noarch/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/t/ja-1ba1ac63-9f8d-450e-a469-83216f398a2f/nanshe/osx-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/t/ja-1ba1ac63-9f8d-450e-a469-83216f398a2f/nanshe/noarch/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): repo.continuum.io
.DEBUG:dotupdate:fetching repodata: args ('https://repo.continuum.io/pkgs/free/osx-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://repo.continuum.io/pkgs/free/noarch/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://repo.continuum.io/pkgs/pro/osx-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://repo.continuum.io/pkgs/pro/noarch/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/t/ja-1ba1ac63-9f8d-450e-a469-83216f398a2f/jakirkham/osx-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/t/ja-1ba1ac63-9f8d-450e-a469-83216f398a2f/jakirkham/noarch/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/t/ja-1ba1ac63-9f8d-450e-a469-83216f398a2f/javascript/osx-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}
.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/t/ja-1ba1ac63-9f8d-450e-a469-83216f398a2f/javascript/noarch/',) kwargs {'session': <conda.connection.CondaSession object at 0x10574c410>, 'use_cache': False}

INFO:stdoutlog:

Solving package specifications: INFO:stdoutlog:Solving package specifications: 
.DEBUG:dotupdate:Solving for [MatchSpec('numpy'), MatchSpec('python 2.7*')]
.DEBUG:dotupdate:Checking satisfiability
.DEBUG:dotupdate:Package removal metric: 0
.DEBUG:dotupdate:Initial package version/build metrics: 0/0
.DEBUG:dotupdate:Track feature count: 1
.DEBUG:dotupdate:Package feature count: 1
.DEBUG:dotupdate:Dependency update count: 0
.DEBUG:dotupdate:Additional package version/build metrics: 1/0
.DEBUG:dotupdate:Weak dependency count: 7
.DEBUG:dotupdate:Looking for alternate solutions

INFO:stdoutlog:


Package plan for installation in environment /zopt/conda2/envs/numpyenv:

The following NEW packages will be INSTALLED:

    blas:            1.1-openblas                  conda-forge
    ca-certificates: 2016.8.2-3                    conda-forge
    libgfortran:     3.0.0-0                       conda-forge
    ncurses:         5.9-8                         conda-forge
    numpy:           1.11.1-py27_blas_openblas_201 conda-forge [blas_openblas]
    openblas:        0.2.18-4                      conda-forge
    openssl:         1.0.2h-2                      conda-forge
    python:          2.7.12-1                      defaults   
    readline:        6.2-0                         conda-forge
    sqlite:          3.13.0-1                      conda-forge
    tk:              8.5.18-0                      defaults   
    zlib:            1.2.8-3                       conda-forge

Linking packages ...
INFO:print:Linking packages ...
INFO:progress.start:12
INFO:progress.update:('ca-certificates', 0)
INFO:progress.update:('libgfortran', 1)
INFO:progress.update:('ncurses', 2)
INFO:progress.update:('sqlite', 3)
INFO:progress.update:('tk', 4)
INFO:progress.update:('zlib', 5)
INFO:progress.update:('openblas', 6)
INFO:progress.update:('openssl', 7)
INFO:progress.update:('readline', 8)
INFO:progress.update:('blas', 9)
INFO:progress.update:('python', 10)
INFO:progress.update:('numpy', 11)
[      COMPLETE      ]|################################################################################################################################| 100%
INFO:progress.stop:None
#
# To activate this environment, use:
# $ source activate numpyenv
#
# To deactivate this environment, use:
# $ source deactivate
#
```

**Comment by kalefranz** [comment]
@msarahan What's the ETA for the next conda-build release?

**Comment by msarahan** [comment]
1.21.10 is already released with this fix.  That carries the question of how to build it, though - @ilanschnell has not responded to me about the entry points, and you have not responded to me about #3224, which would fix entry points for the way that 1.21.10 does them (with setuptools/pip).

Either I need to revert to distutils + old style entry points and spend an undefined amount of time debugging, or you need to get #3224 out there.

**Comment by msarahan** [comment]
More context: https://github.com/conda-forge/pyferret-feedstock/pull/8#issuecomment-236740370 highlights the issue I was seeing with entry points.  This is what I'm waiting on feedback from @ilanschnell on.  The move to setuptools/pip entry points was/is my workaround for that issue.

**Comment by ilanschnell** [comment]
I looked at https://github.com/conda-forge/pyferret-feedstock/pull/8#issuecomment-236740370 and still don't understand the problem.  @msarahan can you explain (in ideally one sentence) what the problem is?  And how moving to different entry points fixes it.

**Comment by msarahan** [comment]
@ilanschnell this is the problem:

```
+ conda inspect linkages -n _test pyferret
usage: conda-inspect [-h] [-c] [-f] [-q] [--no-remove] [dir [dir ...]]
conda-inspect: error: unrecognized arguments: -n _test pyferret
```

The conda-inspect entry point is clearly messed up.  It is missing several arguments that should be there (https://github.com/conda/conda-build/blob/master/conda_build/main_inspect.py#L52-L89).  The conda-build version in use here is 1.21.9, as built at conda-forge with the recipe at https://github.com/conda-forge/conda-build-feedstock/blob/master/recipe/meta.yaml

Going to a different entry point (using setuptools) makes that entry point actually work with all of the arguments to the ```linkages``` subcommand.

You currently build conda-build and conda with an internal recipe, using your legacy system, correct?  That is a key difference here - the entry points here may have always been broken with conda-build, but we just didn't know because we were using your internal build system rather than conda-build to build conda-build packages.

**Comment by ilanschnell** [comment]
The entry points in the internal system work the same as in conda-build.  On Windows, there is the binary executable, which basically executes `python.exe` in the directory above and passes all the arguments to it.  I made a small experiment.  I took the executable, called it `foo.exe`, and created `foo-script.py` with the following content:
```
import sys
print(sys.argv)
```
The output is as expected when calling `foo inspect linkages -n _test pyferret`.  Is it possible that the bug is in conda?  Because `conda inspect linkages -n _test pyferret` should simply call `conda-inspect linkages -n _test pyferret`.  What happens when you instead of invoking `conda inspect linkages -n _test pyferret`, you invoke `conda-inspect linkages -n _test pyferret` directly?

**Comment by ilanschnell** [comment]
Also, being on `1.21.10`, I get (for example):
```
> conda inspect linkages -n _test xlwings
Error: conda inspect linkages is only implemented in Linux and OS X
```

**Comment by msarahan** [comment]
@ilanschnell, thanks for looking into this, and sorry for the confusion.  There are a few issues at play here.

1) The pyferret issue occurs on Linux, not on Windows.  I understand what the behavior *should* be, but it seems the subcommand stuff isn't working the way that it should.  That error message I copied here shows that the entry point is working at a fundamental level, and that argparse is working to some extent, but the subcommand to do the linkages inspection is not working.

2) Windows is a separate issue, brought on by my alternative approach to use setuptools entry points, installed by pip.  I wanted setuptools entry points only as something to try other than conda/conda-build entry points.  I switched to them when I was able to make conda-build work with them.  I know that the .pth mess that setuptools entry points create is not desirable at all.  That's why I use Pip to install these packages (rather than ```python setup.py install```.  It takes the setuptools entry point information, but creates sane entry points like conda/conda-build do.

**Comment by kalefranz** [comment]
Now that conda-build 1.21.10 is released, I think I can close this issue.

**Comment by ilanschnell** [comment]
Well conda-build 1.21.10 is tagged, but there are no packages in the default channel yet.

@msarahan it seems like the problem is on the argparse level, and that using pip entry points is just a workaround for the real problem.  It is OK if I patch the `setup.py` back the way it was before, in order to create the conda-build 1.21.10 packages?

**Comment by msarahan** [comment]
@ilanschnell sure, that's a totally fine solution to get packages out ASAP.  We can talk about and debug the entry point issue more later.

**Comment by jakirkham** [comment]
Sorry I feel like I have missed a few steps here. This build of `conda` is from `defaults` and to my knowledge is not built with the `conda-forge` copy of `conda-build`. Also I'm failing to see how an issue with `conda-build` is affecting a pure `conda` operation. Could someone please help me understand what I'm missing?
